core/store/src/trie/shard_tries.rs

```diff
@@ -379,6 +383,35 @@ impl ShardTries {
         remove_all_state_values(store_update, shard_uid);
     }
 
+    /// Remove trie from memory for shards not included in the given list.
+    pub fn retain_mem_tries(&self, shard_uids: &[ShardUId]) {
+        self.0.mem_tries.write().unwrap().retain(|shard_uid, _| shard_uids.contains(shard_uid));
+    }
+
+    /// Remove trie from memory for given shard.
+    pub fn unload_mem_trie(&self, shard_uid: &ShardUId) {
+        self.0.mem_tries.write().unwrap().remove(shard_uid);
+    }
+
+    /// Loads in-memory-trie for given shard and state root (if given).
+    pub fn load_mem_trie(
+        &self,
+        shard_uid: &ShardUId,
+        state_root: Option<StateRoot>,
+    ) -> Result<(), StorageError> {
+        let mem_tries = load_trie_from_flat_state_and_delta(&self.0.store, *shard_uid, state_root)?;
+        self.0.mem_tries.write().unwrap().insert(*shard_uid, Arc::new(RwLock::new(mem_tries)));
+        Ok(())
+    }
+
+    /// Should be called on upon catchup to decide whether to load mem-trie for the shard.
+    pub fn should_load_mem_trie_on_catchup(&self, shard_uid: &ShardUId) -> bool {
```

Comment:
Function seems redundant. Can combine the functionality with `load_mem_trie`

Reply:
`load_mem_trie` is used in another place too.
Now we have `is_mem_trie_loaded` instead of `should_load_mem_trie_on_catchup` and I believe it makes sense now to have this function.